src/bindings/xfconf/elektra-xfconf-channel.c

```diff
@@ -0,0 +1,1121 @@
+ * @param channelName the channel name to convert to, must be a valid pointer
+ * @return a new pointer to a string with the elektra key name, must be freed by the caller
+ */
+static char * channelNameToKeyName (const char * channelName)
+{
+	char * kdbKeyName = malloc ((strlen (channelName) + strlen (XFCONF_ROOT) + strlen (XFCONF_NAMESPACE) + 3) * sizeof (char));
+	sprintf (kdbKeyName, "%s%s/%s", XFCONF_NAMESPACE, XFCONF_ROOT, channelName);
+	return kdbKeyName;
+}
+
+static KeySet * keySet_from_channel (const gchar * channel_name, uint reference, uint already_locked)
+{
+	trace ();
+	char * kdbKeyName;
+	if (XFCONF_INDIVIDUAL_KEY_SETS == 1)
+	{
+		kdbKeyName = channelNameToKeyName (channel_name);
+	}
+	else
+	{
+		kdbKeyName = channelNameToKeyName ("");
+	}
+	ChannelKeySetPair * channelPair = find_or_create_channel_pair (channel_name, 1);
+	Key * parentKey = keyNew (kdbKeyName, KEY_END);
+	g_debug ("Fetch keys from parent: %s", kdbKeyName);
+	if (already_locked != 1)
+	{
+		require_channel_write_lock ()
+	}
+	g_debug ("Refreshing keyset for channel %s from kdb, currently %ld keys", channel_name, ksGetSize (channelPair->keySet));
+	int getStatusCode = kdbGet (elektraKdb, channelPair->keySet, parentKey);
+	g_debug ("Refreshed keyset (status code was %d) now has %ld keys", getStatusCode, ksGetSize (channelPair->keySet));
+	if (already_locked != 1)
+	{
+		release_channel_lock ()
+	}
+
+	switch (getStatusCode)
+	{
+	case -1:
+		g_warning ("There was a failure fetching the keys");
+		break;
+	case 0:
+		g_debug ("The keyset did not change");
+		break;
+	case 1:
+		g_debug ("Retrieved the keyset");
+		break;
+	case 2:
+		g_debug ("No data was loaded from storage");
+		break;
+	default:
+		g_warning ("An unknown status code(%d) occurred during keyset fetch", getStatusCode);
+	}
+	if (XFCONF_DEBUG_LOG_FOUND_KEYS == 1)
+	{
+		Key * cur;
+		if (already_locked != 1)
+		{
+			require_channel_read_lock ()
+		}
+		ssize_t keySetSize = ksGetSize (channelPair->keySet);
+		g_debug ("KeySet has %ld keys", keySetSize);
+		for (elektraCursor i = 0; i < keySetSize; i++)
+		{
+			cur = ksAtCursor (channelPair->keySet, i);
+			g_debug ("Found key: %s", keyName (cur));
+		}
+		if (already_locked != 1)
+		{
+			release_channel_lock ()
+		}
+		free (kdbKeyName);
+	}
+	if (reference != 0)
+	{
+		return channelPair->keySet;
+	}
+	if (already_locked != 1)
+	{
+		require_channel_read_lock ()
+	}
+	KeySet * deepCopy = ksDeepDup (channelPair->keySet);
+	if (already_locked != 1)
+	{
+		release_channel_lock ()
+	}
+	return deepCopy;
+}
+
+/**
+ * Convert a string to a GValue - This function tries to parse the content of `new_value` and stores it in the `g_value` pointer.
+ * If that is not possible, e.g. the `g_value` has an unknown type, this function does nothing.
+ * The desired type is taken from the `g_value`.
+ * @param g_value the value to parse the string to - not null
+ * @param new_value the string of the new value - not null
+ * @return 0 on success and 1 otherwise
```

Comment:
```suggestion
 * @retval 0 on success and 1 otherwise
```

Reply:
Should be done by 6506d0e23efa25047381ef0cd2afa5aed13da3f7 or before